ui/src/pageLayout/components/CloudNav.tsx

```diff
@@ -1,63 +1,101 @@
-import React, {PureComponent} from 'react'
+// Libraries
+import React, {FunctionComponent} from 'react'
+import {Link} from 'react-router'
+import {connect} from 'react-redux'
 
 // Components
 import {FeatureFlag} from 'src/shared/utils/featureFlag'
-import {NavMenu, Icon} from '@influxdata/clockface'
+import {
+  AppHeader,
+  PopNav,
+  Button,
+  ComponentColor,
+  FlexBox,
+  FlexDirection,
+  ComponentSize,
+} from '@influxdata/clockface'
 import CloudOnly from 'src/shared/components/cloud/CloudOnly'
+
+// Constants
 import {
   CLOUD_URL,
   CLOUD_USAGE_PATH,
   CLOUD_BILLING_PATH,
+  CLOUD_SIGNOUT_URL,
 } from 'src/shared/constants'
 
 // Types
-import {IconFont} from '@influxdata/clockface'
-
-export default class CloudNav extends PureComponent {
-  render() {
-    return (
-      <CloudOnly>
-        <NavMenu.Item
-          active={false}
-          titleLink={className => (
-            <a className={className} href={this.usageURL}>
-              Usage
-            </a>
-          )}
-          iconLink={className => (
-            <a className={className} href={this.usageURL}>
-              <Icon glyph={IconFont.Cloud} />
-            </a>
-          )}
-        >
-          <FeatureFlag name="cloudBilling">
-            <NavMenu.SubItem
+import {AppState} from 'src/types'
+
+// Images
+import Logo from '../images/influxdata-logo.png'
+
+interface StateProps {
+  orgName: string
+  orgID: string
+}
+
+const CloudNav: FunctionComponent<StateProps> = ({orgName, orgID}) => {
+  const usageURL = `${CLOUD_URL}${CLOUD_USAGE_PATH}`
+  const billingURL = `${CLOUD_URL}${CLOUD_BILLING_PATH}`
+  const handleUpgradeClick = (): void => {
+    window.location.assign(billingURL)
```

Comment:
why doesn't the router doesn't take care of this?

Reply:
as in using `router.push`?

Reply:
yus